### PR TITLE
Keep track of running jobs

### PR DESCRIPTION
**Pull request by hellais**
When a job is deleted remove it from the list of runningJobs and mark it
as done.
This fixes the bug spotted by @sbs and @lorenzoPrimi

### CODE REVIEW

File: proteus-events/events/scheduler.go [hunk 1]
```diff
@@ -691,6 +702,7 @@ func (s *Scheduler) Start() {
 		return
```
Comment (bassosimone) [inline proteus-events/events/scheduler.go]: I think the `XXX` comment above is not accurate anymore, right?

File: proteus-events/events/events.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func ListJobs(db *sqlx.DB, showDeleted bool) ([]JobData, error) {
 
 var ErrJobNotFound = errors.New("job not found")
 
-func DeleteJob(jobID string, db *sqlx.DB) (error) {
+func DeleteJob(jobID string, db *sqlx.DB, s *Scheduler) (error) {
```
Comment (bassosimone) [inline proteus-events/events/events.go]: I don't understand why you're passing the scheduler to this function. Where you use it?

Comment (bassosimone) [inline proteus-events/events/events.go]: lol, okay. I probably need more coffee :-).

**Comment by bassosimone** [comment]
I am looking into this PR.

**Comment by hellais** [comment]
> the s.runningJobs map is only used to check whether one is trying to delete a job more than once.

Actually this is needed to keep a pointer to the jobs instances, otherwise we would not have any way of addressing the `IsDone` key.

> If this understanding is correct, then the PR is correct.

A part from that, yes I think you got it :)

**Pull request merged**